adagrams/game.py

```diff
@@ -1,11 +1,97 @@
+from random import choice
+from copy import copy
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    user_hand = []
+    available_letters = []
+
+    for letter in LETTER_POOL:
+        for num in range(0, LETTER_POOL[letter]):
+            available_letters.append(letter)
+    while len(user_hand) < 10:
+        letter_choice = choice(available_letters)
+        removed_letter = available_letters.index(letter_choice)
+        user_hand.append(available_letters.pop(removed_letter))
+    return user_hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    user_hand = copy(letter_bank)
+    for letter in word.upper():
+        if letter not in user_hand:
+            return False
+        else:
+            letter_index = user_hand.index(letter)
+            user_hand.pop(letter_index)
+    return True
 
 def score_word(word):
-    pass
+    score = 0
+    points_dict = {
+    1 : ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"], 
+    2: ["D", "G"],
+    3: ["B", "C", "M", "P"],
+    4: ["F", "H", "V", "W", "Y"],
+    5: ["K"],
+    8: ["J", "X"],
+    10: ["Q", "Z"]
+    } 
+
+    for letter in word.upper():
+        for key in points_dict:
+            if letter in points_dict[key]:
```

Comment:
Here when we look for `letter` in `points_dict[key]` we're looking through a list so time complexity is linear. 

You could refactor score_word like this to avoid using `in` with a list and also only use a single for loop: 
```python
def score_word(word):
    score = 0
    for letter in word.upper():
        score += SCORE_CHART[char]
    if len(word) > 6:
        score += 8
    return score